dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -222,7 +222,7 @@ public WebMessage updateIcon(@PathVariable String key, HttpServletRequest reques
   public WebMessage deleteIcon(@PathVariable String key)
       throws NotFoundException, BadRequestException {
 
-    iconService.deleteIcon(key);
+    iconService.deleteIcon(key.trim());
```

Comment:
Why are you doing that? Can we not expect users to give us the exact key they want to delete/search for?

I would only validate on insert that they key is valid i.e. has on whitespace.

Reply:
handling unintended whitespace in user input

Reply:
Whitespace has to be percent encoded. If a user goes to that length it would result in a 404 as the key does not exist.

If you try adding spaces in a URL with a tool like curl will yell at you

```
curl 'http://admin:district@localhost:8080/api/tracker/events/ FV4JCI73wO2'
curl: (3) URL using bad/illegal format or missing URL
```

if you percent encode it you will get a 400 as in this particular case its expected to be a UID

```
curl 'http://admin:district@localhost:8080/api/tracker/events/%20FV4JCI73wO2'
{"httpStatus":"Bad Request","httpStatusCode":400,"status":"ERROR","message":"Value ' FV4JCI73wO2' is not valid for path parameter uid. UID must be an alphanumeric string of 11 characters starting with a letter."}
```

If you select the above url and right-click go in Chrome it will percent encode the space and you also get a 400.

We should not add code that does not add any benefit.